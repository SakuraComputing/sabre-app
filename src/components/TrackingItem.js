import React from 'react';
import moment from 'moment';

const TrackingItem = ({ status, trackDate, serviceCentre, tour, comment } ) => (
    <tr>
        <td>{status}</td>
        <td>{moment(trackDate).format('dddd, MMMM Do YYYY, h:mm:ss a')}</td>
        <td>{serviceCentre}</td>
        <td>{tour}</td>
        <td>{comment}</td>
    </tr>
);

export default TrackingItem;

