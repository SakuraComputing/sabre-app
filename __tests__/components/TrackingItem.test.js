import React from 'react';
import { shallow } from 'enzyme';
import TrackingItem from '../../src/components/TrackingItem';

const track = {
    status: "Out for Delivery",
    parcelId: "JD0002256504047335",
    trackDate: 1510834763,
    serviceCentre: "Borehamwood",
    tour: "BE45",
    comment: "On the van"
};

describe('when tracking item renders',() => {
    test('it should render a single record correctly',() => {
        const wrapper = shallow(<TrackingItem tracks={track}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
