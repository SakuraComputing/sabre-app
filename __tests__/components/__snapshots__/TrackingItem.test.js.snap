// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when tracking item renders it should render a single record correctly 1`] = `
<tr>
  <td />
  <td>
    Thursday, January 1st 1970, 12:00:00 am
  </td>
  <td />
  <td />
  <td />
</tr>
`;
